### ISSUE TITLE
Add overlap option for cone search 

### ISSUE THREAD

**Issue by cdeil** [astropy/astropy-healpix#67]
In Gammapy we currently call [healpy.query_disc](http://healpy.readthedocs.io/en/latest/generated/healpy.query_disc.html) with both the `inclusive=True` and the `inclusive=False` option (see https://github.com/gammapy/gammapy/pull/1167#issuecomment-337156775).

As discussed in https://github.com/astropy/astropy-healpix/issues/41 , probably we don't want to spend the time to fully implement `healpy.query_disc` concerning the `inclusive=False` behaviour.

But maybe we could add an option to [HEALPix.cone_search_lonlat](http://astropy-healpix.readthedocs.io/en/latest/api/astropy_healpix.HEALPix.html#astropy_healpix.HEALPix.cone_search_lonlat) that corresponds to the `inclusive=True` behaviour, i.e. restricts the pixels to the ones where the center is in the cone?

@astrofrog - OK to add this?

How about calling it `overlap={'center', 'full'}` which is maybe a bit more intuitive than `inclusive = {True, False}` and also more extensible should we want to add other options in the future?
Concerning implementation: should it just be done at the end of the function in an `if overlap == 'center'` block using a Numpy expression, subsetting with a mask that's true if the pixel center is in the cone? Or would you suggest this be added in the C or Cython code somehow?

**Comment by astrofrog** [comment]
I agree with this - I would maybe call the options ``overlap={'center', 'any'}`` because ``full`` to me sounds like the pixel has to overlap fully whereas really any overlap will do.

Implementation-wise, doing it in ``core.py`` in Python/Numpy is fine - if we want to avoid making temporary arrays then doing it in Cython would be better, but honestly I think just doing it in ``core.py`` would be fine for now and we can optimize later?

**Comment by tboch** [comment]
> I agree with this - I would maybe call the options overlap={'center', 'any'} because full to me sounds like the pixel has to overlap fully whereas really any overlap will do.

+1 for using 'center' and 'any', sounds easier to understand to me

**Comment by cdeil** [comment]
Agreed that "full" was a complete misnomer and that `overlap="any"` is good.

>  Implementation-wise, doing it in core.py in Python/Numpy is fine - if we want to avoid making temporary arrays then doing it in Cython would be better, but honestly I think just doing it in core.py would be fine for now and we can optimize later?

Agreed. I'll try to find time to make a pull request later this week.
